rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -286,13 +467,340 @@ error_code cellSearchStartContentSearchInList(vm::cptr<CellSearchContentId> list
+						{
+							// Failed to load stream information
+							avformat_free_context(av_format_ctx);
+							continue;
+						}
+
+						// Derive first audio stream id from avformat context
+						int stream_index = -1;
+						for (uint i = 0; i < av_format_ctx->nb_streams; i++)
+						{
+							if (av_format_ctx->streams[i]->codecpar->codec_type == AVMEDIA_TYPE_AUDIO)
+							{
+								stream_index = i;
+								break;
+							}
+						}
+						if (stream_index == -1)
+						{
+							// Failed to find an audio stream
+							avformat_free_context(av_format_ctx);
+							continue;
+						}
+
+						AVStream* stream = av_format_ctx->streams[stream_index];
+						AVCodecParameters* codec = stream->codecpar;
+
+						info.bitrate = codec->bit_rate / 1000;  // TODO: Assumption, verify value
+						info.quantizationBitrate = codec->bit_rate / 1000;  // TODO: Assumption, verify value
+						info.samplingRate = codec->sample_rate;  // TODO: Assumption, verify value
+						info.drmEncrypted = 0; // Needs to be 0 or it wont be accepted
+						info.duration = av_format_ctx->duration / 1000;  // TODO: Assumption, verify value
+						info.releasedYear = 0; // TODO: Use "date" id3 tag for this
+						info.size = item.size;
+						info.playCount = 0; // we do not track this for now
+						info.lastPlayedDate = 0;  // we do not track this for now
+						info.importedDate = 0; // we do not track this for now
+						info.status = CELL_SEARCH_CONTENTSTATUS_AVAILABLE; // CellSearchContentStatus
+
+						switch (codec->codec_id) // AVCodecID
+						{
+						case AV_CODEC_ID_MP3:
+							info.codec = CELL_SEARCH_CODEC_MP3; // CellSearchCodec
+							break;
+						case AV_CODEC_ID_AAC:
+							info.codec = CELL_SEARCH_CODEC_AAC; // CellSearchCodec
+							break;
+						case AV_CODEC_ID_AC3:
+							info.codec = CELL_SEARCH_CODEC_AC3; // CellSearchCodec
+							break;
+						case AV_CODEC_ID_WMAV1:
+						case AV_CODEC_ID_WMAV2:
+							info.codec = CELL_SEARCH_CODEC_WMA; // CellSearchCodec
+							break;
+						case AV_CODEC_ID_ATRAC3:
+							info.codec = CELL_SEARCH_CODEC_AT3; // CellSearchCodec
+							break;
+						case AV_CODEC_ID_ATRAC3P:
+							info.codec = CELL_SEARCH_CODEC_AT3P; // CellSearchCodec
+							break;
+						default:
+							info.codec = CELL_SEARCH_CODEC_UNKNOWN; // CellSearchCodec
+							info.status = CELL_SEARCH_CONTENTSTATUS_NOT_SUPPORTED; // CellSearchContentStatus
+							break;
+						}
+
+						AVDictionaryEntry *tag;
+						std::string value;
+
+						info.trackNumber = 0;
+						tag = av_dict_get(av_format_ctx->metadata, "track", 0, AV_DICT_IGNORE_SUFFIX);
+						if (tag != nullptr)
+						{
+							std::string tmp(tag->value);
+							info.trackNumber = stoi(tmp.substr(0, tmp.find("/")));
+						}
+
+						tag = av_dict_get(av_format_ctx->metadata, "album", 0, AV_DICT_IGNORE_SUFFIX);
+						if (tag != nullptr)
+						{
+							value = tag->value;
+							if (value.size() > CELL_SEARCH_TITLE_LEN_MAX)
```

Comment:
Is there a reason to manually truncate those temporary values if `strcpy_trunc` properly truncates them anyway?

Also applies to the same conditions below.